docs/source/CartesianConfig.rst

```diff
@@ -427,6 +345,169 @@ values. If in doubt of where to define or set a key, placing it at the
+    Dictionary #3:
+        depend = ['B.one', 'B.two']
+        key1 = value1
+        key2 = value2
+        key3 = value3
+        name = B.three
+        shortname = B.three
+
+
+Named variant set
+-----------------
+
+It is possible to assign a name to a variant set.  This enables an entire
+variant set to be used in filters_. All variant set combinations will contain an
+extra key/value pair for each named variant set it contains, the key is
+the variant set name and the value is the corresponding current variant name.
+For example::
+
+   variants var1_name:
+        - one:
+            key1 = Hello
+        - two:
+            key2 = World
+        - three:
+   variants var2_name:
+        - one:
+            key3 = Hello2
+        - two:
+            key4 = World2
+        - three:
+
+   only (var2_name=one).(var1_name=two)
+
+Results in the following when parsed with ``cartesian_config.py -c``::
+
+    dict    1:  (var2_name=one).(var1_name=two)
+          dep = []
+          key2 = World         # variable key2 from variants var1_name and variant two.
+          key3 = Hello2        # variable key3 from variants var2_name and variant one.
+          name = (var2_name=one).(var1_name=two)
+          shortname = (var2_name=one).(var1_name=two)
+          var1_name = two      # variant name in same namespace as variables.
+          var2_name = one      # variant name in same namespace as variables.
+
+Named variants could also be used as normal variables.::
+
+   variants guest_os:
+        - fedora:
+        - ubuntu:
+   variants disk_interface:
+        - virtio:
+        - hda:
+
+Results in the following::
+
+    dict    1:  (disk_interface=virtio).(guest_os=fedora)
+        dep = []
+        disk_interface = virtio
+        guest_os = fedora
+        name = (disk_interface=virtio).(guest_os=fedora)
+        shortname = (disk_interface=virtio).(guest_os=fedora)
+    dict    2:  (disk_interface=virtio).(guest_os=ubuntu)
+        dep = []
+        disk_interface = virtio
+        guest_os = ubuntu
+        name = (disk_interface=virtio).(guest_os=ubuntu)
+        shortname = (disk_interface=virtio).(guest_os=ubuntu)
+    dict    3:  (disk_interface=hda).(guest_os=fedora)
+        dep = []
+        disk_interface = hda
+        guest_os = fedora
+        name = (disk_interface=hda).(guest_os=fedora)
+        shortname = (disk_interface=hda).(guest_os=fedora)
+    dict    4:  (disk_interface=hda).(guest_os=ubuntu)
+        dep = []
+        disk_interface = hda
+        guest_os = ubuntu
+        name = (disk_interface=hda).(guest_os=ubuntu)
+        shortname = (disk_interface=hda).(guest_os=ubuntu)
+
+Implicit keys
+-------------
+
+Some special keys are expected to be defined with a list of values and
+implicitly define a set of keys for each one of its values. For example, each
+value of the 'vms' key will implicitly define keys such as 'mem_<vm>'.
+In this case a default 'mem' key is also available. Example::
```

Comment:
This is a new section, right? I'm not sure I understand it properly, because cartesian doe snot implicitly define any keys, but it allows querying for object params, which overrides values of keys ending with the object's name. For example querying for `mem` returns the value of `mem`, but when querying for `mem` of a `vm1` object, it first looks whether `mem_vm1` exists and reports it's value, otherwise it fall-backs to `mem`.

This does not work just for a single dimension, it can be applied recursively, so one can query for `mem` value of object `vm1` of object `provider1`, which first checks whether `mem_vm1_provider1` exists, then it looks for `mem_vm1` and last it looks for `mem`.

Anyway this is not limited to `mem` only, it works for any param.

PS: For explanation I used the looks for..., but in reality one uses `params = params.object_params("vm1")` which duplicates the `params` and replaces all `_vm1` suffixed values. Anyway the behavior is the same...